### ISSUE TITLE
LR scheduler step() behaviour with and without epoch parameter 

### ISSUE THREAD

**Issue by DianeBouchacourt** [pytorch/pytorch#55135]
I have a question regarding the epoch parameter in LR scheduler. I understand passing epoch as a parameter to self.step() is deprecated, however given 

https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/torch/optim/lr_scheduler.py#L157

If the epoch is not None, then _get_closed_form is called if the scheduler has it, whereas it is not the case when epoch is None.

Doesn't it mean a different behaviour with and without the epoch parameter passed in self.step ?
